sources/ghostdog/src/main.rs

```diff
@@ -160,6 +175,51 @@ fn is_neuron_attached() -> Result<()> {
     }
 }
 
+fn is_infiniband_attached() -> Result<()> {
+    // confirm device has `SW_MNG` in the VPD file
+    let infiniband_sys_devices =
+        fs::read_dir("/sys/class/infiniband/").context(error::InfinibandSysDevicesSnafu)?;
+
+    for device in infiniband_sys_devices {
+        let device = device.context(error::InfinibandSysDevicesSnafu)?;
+        let device_name: OsString = device.file_name().clone();
```

Comment:
nit, `file_name()` already returns an owned `OsString`:

```suggestion
        let device_name: OsString = device.file_name();
```